app/models/conditions_response/condition_responder.rb

```diff
@@ -66,21 +66,40 @@ def self.get_timing(condition)
   end
 
 
-  # Determine the number of days toay is _away_from_ this date.
+  # Determine the number of days today is _away_from_ this date.
   # The :timing is whether we are calculating the number of days
   # _before_, _after_, or _on_ this day, starting from today.
   #
   # @param this_date [Date] - the date to compare to today
   # @param timing [Timing] - which 'direction' (before, after, on) to compare to today
   # @return [Integer] - the number of days away from today, based on our :timing
   def self.days_today_is_away_from(this_date, timing)
+    days_1st_date_is_from_2nd(Date.current, this_date, timing)
+  end
+
+
+  # Determine the number of days :a_date is _away_from_ :second_date.
+  # The :timing is whether we are calculating the number of days
+  # _before_, _after_, or _on_ this day, starting from :a_date.
+  #
+  # @param a_date [Date] - the starting date
+  # @param second_date [Date] - the date to compare to :a_date
+  # @param timing [Timing] - which 'direction' (before, after, on) to compare to :a_date
+  # @return [Integer] - the number of days away from today, based on our :timing
+  def self.days_1st_date_is_from_2nd(a_date, second_date, timing)
 
-    day_num_to_check = 0 # equivalent to checking on today
+    day_num_to_check = 0 # default value
+
+    # We use .to_date to ensure that we're comparing and working with Dates, not Times, etc.
+    # If calling .to_date throws an exception, that's an exception that should be raised.
 
     if timing_is_before?(timing)
-      day_num_to_check = this_date - Date.current
+      # number of days that a_date is _before_ second_date
+      day_num_to_check = second_date.to_date - a_date.to_date
+
     elsif timing_is_after?(timing)
-      day_num_to_check = Date.current - this_date
+      # number of days that a_date is _after_ second_date
+      day_num_to_check = a_date.to_date - second_date.to_date
```

Comment:
The signature here seems needlessly complex.  That is, for the sender of this message, if we're dealing with `:after` timing then that first date arg is the later date, and the second is the earlier date, and vice-versa for `:before` timing.   That should not be the concern of the caller.  And if the caller has to execute the logic to arrange the date args in the correct order, then what value does the method provide?  The caller might as do the simple one-line statement to determine dates difference (using the correct math of `today - earlier_date` or `later_date - today`.), since that logic is required anyway to present the dates in the correct order here.

What am I missing?

Reply:
If the caller is just concerned about `today`, then the right method to use is still `days_today_is_away_from`  That hasn't changed.

This is a more generalized method.  We have discovered that we need to be able to know _what alerts will be going out on what day?_   Because you and I have spent time writing queries on the production system to answer those questions.  QED we need that functionality.  In doing that we were acting in sort of an 'administrator' role.  IOW, we were doing more than just testing.  Susanna was also interested in that question.  (What members will be getting emails tomorrow? or the day after? )

Now that we have more than 1 use case for sending out alerts, we have shown enough need  to answer that question ('what alerts will be going out not just today, but on specific days?').  Thus this method is warranted.

I put together a quick chart on the admin dashboard to help me understand the data. IOW, as I developed the feature, I needed a way to see the information.  That's  a strong indicator that the feature/ability/use case is needed.

<img width="803" alt="dashboard draft-2019-01-11" src="https://user-images.githubusercontent.com/673794/51059850-74349880-15a2-11e9-80ce-56ec68b6debc.png">

_(I don't know if we'll end up with something like this or not.  But it was helpful to me.)_